mllib/src/main/scala/org/apache/spark/mllib/evaluation/MultilabelMetrics.scala

```diff
@@ -38,76 +40,62 @@ class MultilabelMetrics @Since("1.2.0") (predictionAndLabels: RDD[(Array[Double]
       (r.getSeq[Double](0).toArray, r.getSeq[Double](1).toArray)
     })
 
-  private lazy val numDocs: Long = predictionAndLabels.count()
+  /**
+   * Use MultilabelSummarizer to calculate all summary statistics of predictions
+   * and labels on one pass.
+   */
+  private lazy val summary: MultilabelSummary = {
```

Comment:
I also wonder whether it's useful to make this lazy... can you really do anything with the class without computing this?

Reply:
Just because all other impls make summary metrics lazy. Ok  I will make this not lazy.